### PR TITLE
Add agat sp statistics

### PR DESCRIPTION
**Pull request by Leila011**
## Description

add agat_sp_statistics

## Issue ticket number

Closes #xxxx 

## Checklist before requesting a review

- [x] I have performed a self-review of my code

- [x] Conforms to the [Contributing guidelines](https://github.com/viash-hub/base/blob/main/CONTRIBUTING.md)

- [x] Proposed changes are described in the [CHANGELOG.md](https://github.com/viash-hub/base/blob/main/CHANGELOG.md)

- [x] I have tested my code with `viash ns test --parallel -q <name or namespace>`

- Check the correct box. Does this PR contain:
  - [ ] Breaking changes
  - [x] New functionality
  - [ ] Major changes
  - [ ] Minor changes
  - [ ] Documentation
  - [ ] Bug fixes

### CODE REVIEW

**Comment by Leila011** [comment]
**How to deal with an argument that can take multiple type?**
The argument gs can take either a file either a number, so I gave it type string but I am not sure if that is the best way to handle it.
```
     - name: --gs
        alternatives: [-f, -g]
        description: |
          This option inform about the genome size in order to compute more statistics. 
          You can give the size in Nucleotide or directly the fasta file.
        type: string
```

**Comment by jakubmajercik** [comment]
> **How to deal with an argument that can take multiple type?** The argument `gs` can take either a file either a number, so I gave it type string but I am not sure if that is the best way to handle it.
> 
> ```
>      - name: --gs
>         alternatives: [-f, -g]
>         description: |
>           This option inform about the genome size in order to compute more statistics. 
>           You can give the size in Nucleotide or directly the fasta file.
>         type: string
> ```

I would split it into 2 arguments.

File: src/agat/agat_sp_statistics/help.txt [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+     - a clear description,
+     - as much relevant information as possible,
+     - the command used,
+     - a data sample,
+     - an explanation of the expected behaviour that is not occurring.
+
+  Do you want to contribute?:
+    You are very welcome, visit this address for the Contributing
+    guidelines:
+    https://github.com/NBISweden/AGAT/blob/master/CONTRIBUTING.md
```
Comment (jakubmajercik) [inline src/agat/agat_sp_statistics/help.txt:82]: Can we remove this?
```suggestion
```

File: src/agat/agat_sp_statistics/script.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+[[ "$par_d" == "false" ]] && unset par_d
+
+# run agat_sp_statistics
+agat_sp_statistics.pl \
+  -i "$par_gff" \
+  -o "$par_output" \
+  ${par_plot:+-d} \
+  ${par_gs:+--gs "${par_gs}"} \
+  ${par_verbose:+--verbose "${par_verbose}"} \
+  ${par_config:+--config "${par_config}"}
```
Comment (jakubmajercik) [inline src/agat/agat_sp_statistics/script.sh:18]: Add the split `gs` arguments

**Review by emmarousseau — APPROVED** [review]
Reduced test data size
Add tool name to version format
Minor config formatting changes

**Review by rcannood — APPROVED** [review]
Tysm!

**Pull request merged**